Fix many compiler warnings

DePasqualeOrg opened the pull request:
## Proposed changes

There were around 180 compiler warnings, which I've reduced to ~106~ 10. The remaining ones are due to concurrency issues that are more difficult to fix. I think these should be addressed separately. The large number of concurrency warnings probably led to many other warnings being overlooked, which can cause code quality to suffer.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the necessary documentation (if needed)

davidkoski commented on Libraries/MLXLLM/LLMModel.swift:
```diff
@@ -24,12 +24,12 @@ extension LLMModel {
         let prefillStepSize = windowSize ?? 512
         var y = input.text
-        var state: LMOutput.State? = nil
+        let state: LMOutput.State? = nil
```
this seems odd -- a `let` initialized to `nil` can probably be deleted?

DePasqualeOrg commented on Libraries/MLXLMCommon/Adapters/LoRA/LoRAContainer.swift:
```diff
@@ -6,7 +6,7 @@
 
 import Foundation
-import MLX
+@preconcurrency import MLX
```
Okay, I've removed this for MLX. If there's an issue, it can be resolved later with the other concurrency warnings.

DePasqualeOrg commented:
I already left three comments in that thread with the same explanation, but maybe they're not showing on your end:

> perLayerQuantization happens in the block above the part that I deleted, and the compiler says to use perLayerQuantization instead, so I believe the part I deleted is redundant.

Take a look at the full context of that part, and I think it will be clear.

DePasqualeOrg commented:
Formatted ✅
